tools/build_executable/build_executable.py

```diff
@@ -0,0 +1,177 @@
+
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env}; "
+
+    elif sys.platform == "linux":
+        target_platform  = "Linux"
+        run_uat_script   = os.path.realpath(os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh"))        
+        archive_dir      = os.path.realpath(os.path.join(args.output_dir, f"SpearSim-{target_platform}-{build_config}"))
+        config_file_src  = os.path.realpath(os.path.join(os.path.dirname(__file__), "LinuxEngine.ini"))
+        config_file_dest = os.path.realpath(os.path.join(unreal_project_dir, "Config", "Linux", "LinuxEngine.ini"))
+        paks_dir_dest    = os.path.realpath(os.path.join(archive_dir, "LinuxNoEditor", "SpearSim", "Content", "Paks"))
+        unreal_tmp_dir   = ""
+
+        if args.conda_script:
+            conda_script = args.conda_script
+        else:
+            # see https://docs.anaconda.com/anaconda/user-guide/faq
+            conda_script = os.path.expanduser(os.path.join("~", "anaconda3", "etc", "profile.d", "conda.sh"))
+
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env}; "
+
+    else:
+        assert False
+
+    # once we know target_platform, set our default pak file src and dest
+    pak_file_src  = os.path.realpath(os.path.join(args.paks_dir, "kujiale_0000_" + target_platform + ".pak"))
+    pak_file_dest = os.path.realpath(os.path.join(paks_dir_dest, "kujiale_0000_" + target_platform + ".pak"))
+
+    # We need to remove this temp dir (created by the Unreal build process) because it contains paths from previous builds.
+    # If we don't do this step, we will get many warnings during this build:
+    #     Warning: Unable to generate long package name for path/to/previous/build/Some.uasset because FilenameToLongPackageName failed to convert
+    #     'path/to/previous/build/Some.uasset'. Attempt result was '../../../../../../path/to/previous/build/path/to/previous/build/Some', but the
+    #     path contains illegal characters '.'
+    if os.path.exists(unreal_tmp_dir):
+        print(f"[SPEAR | build_executable.py] Unreal Engine cache directory exists, removing: {unreal_tmp_dir}")
+        shutil.rmtree(unreal_tmp_dir)
+
+    if not args.skip_clone_github_repo:
+        
+        # remove our temporary repository directory to ensure a clean build
+        if os.path.exists(repo_dir):
+            print(f"[SPEAR | build_executable.py] Repository exists, removing: {repo_dir}")
+            shutil.rmtree(repo_dir)
+            os.makedirs(repo_dir)
+
+        # clone repo with submodules
+        cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+        print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+
+        # reset to a specific commit ID
+        if args.commit_id:
+            cmd = ["git", "reset", "--hard", args.commit_id]
```

Comment:
Just realized that this doesn't work as intended. It changes the commit in the wrong location. We need to change the commit in `repo_dir`.

```python
if args.commit_id:
    cwd = os.getcwd()
    os.chdir(repo_dir)
    cmd = ["git", "reset", "--hard", args.commit_id]
    print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
    subprocess.run(cmd, check=True)
    os.chdir(cwd)
```

Maybe we need to change the commit in both locations?

Reply:
Good catch.